openhands/llm/retry_mixin.py

```diff
@@ -0,0 +1,85 @@
+from tenacity import (
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_exponential,
+)
+
+from openhands.core.exceptions import OperationCancelled
+from openhands.core.logger import openhands_logger as logger
+from openhands.runtime.utils.shutdown_listener import should_exit
+
+
+class RetryMixin:
+    """Mixin class for retry logic."""
+
+    def retry_decorator(self, **kwargs):
+        """
+        Create a retry decorator with customizable parameters.
+
+        Args:
+            **kwargs: Keyword arguments to override default retry behavior.
+                      Keys: num_retries, retry_exceptions, retry_min_wait, retry_max_wait, retry_multiplier
+
+        Returns:
+            A retry decorator with the specified or default parameters.
+        """
+        num_retries = kwargs.get('num_retries')
+        retry_exceptions = kwargs.get('retry_exceptions')
+        retry_min_wait = kwargs.get('retry_min_wait')
+        retry_max_wait = kwargs.get('retry_max_wait')
+        retry_multiplier = kwargs.get('retry_multiplier')
+
+        def custom_completion_wait(self, retry_state):
+            """Custom wait for completion."""
+            exception = retry_state.outcome.exception() if retry_state.outcome else None
+            if exception is None:
+                return 0
+
+            # retry_min_wait = config.retry_min_wait
+            # retry_max_wait = config.retry_max_wait
+
+            # rate limit errors
+            exception_type = type(exception).__name__
+            logger.error(rf'\exception_type: {exception_type}\n')
```

Comment:
I think `log_retry_attempt` does this logging of an error, here: https://github.com/All-Hands-AI/OpenHands/pull/4057/files#diff-58cd3ae5a235b45a1042b4e1ed86ceff2f43afc13b145bd173b39d94e7e58108R82

Reply:
Can do without this logger, sure.